content/tutorials/how-to-ably-react-hooks.textile

```diff
@@ -0,0 +1,221 @@
+
+```[javascript]
+const [channel] = useChannel("[?rewind=100]your-channel-name", (message) => {
+    // List the last 100 messages on the channel
+    console.log(message);
+});
+```
+
+h2. Step 6 - List present members
+
+The @usePresence@ hook lets you subscribe to presence messages on a channel, so that you can be notified when members enter or leave the channel and retrieve the current presence set.
+
+h3. Retrieving the Presence set
+
+To use the Presence capability, you need to assign each client a unique "@clientId@":/api/realtime-sdk#client-options. In a production application, this @clientId@ should be returned by your token request endpoint (see Step 3). But you can also do this in the call to @configureAbly@:
+
+```[javascript]
+configureAbly({ key: "your-ably-api-key", clientId: generateRandomId() });
+```
+
+You can then import and use the @usePresence@ hook. It returns an array of standard Ably "@PresenceMessage@":/api/realtime-sdk/presence#presence-message objects:
+
+```[javascript]
+import { configureAbly, useChannel, usePresence } from "@ably-labs/react-hooks";
+
+const [presenceData] = usePresence("channel-name", "optional initial state");
+
+// Convert presence data to list items to render    
+const members = presenceData.map((msg, index) => <li key={index}>{msg.clientId}: {msg.data}</li>);
+```
+
+h3. Updating the status of a channel member
+
+You can update presence data using the @updateStatus@ function:
+
+```[javascript]
+const [presenceData, updateStatus] = usePresence("your-channel-name", "optional initial state");
+
+updateStatus("new status");
+```
+
+The new state will be sent to the channel, and any other clients subscribed to the channel will be notified of the change immediately.
+
+If you don't want to use the presence data returned from @usePresence@, you can configure a callback:
+
+```[javascript]
+const [_, updateStatus] = usePresence("channel-name", "optional initial state", (presenceUpdate) => {
+    console.log(presenceUpdate);
+});
+```
+
+The @usePresence@ hook supports objects, as well as strings:
+
+```[javascript]
+usePresence("your-channel-name", { someProperty: "someValue" });
+```
+
+h3. Using type hints (TypeScript only)
+
+If you are using TypeScript, there are type hints to ensure that presence updates are of the same type as your initial constraint, or a provided generic type parameter. In this example, the @MyPresenceType@ is checked. If it is omitted, the shape of the initial value will be used. If that's omitted, the default will be of type @any@:
+
+```[typescript]
+const TypedUsePresenceComponent = () => {
+    const [val] = usePresence<MyPresenceType>("testChannelName", { foo: "bar" });
+
+    return (
+        <div role='presence'>
+            {JSON.stringify(val)}
+        </div>
+    );
+}
+
+interface MyPresenceType {
+    foo: string;
+}
+```
+
+h2. 
```

Comment:
Looks good under web view, but it does end abruptly and I'm wondering if there's a section missing here? Perhaps there should be further info, including a link to the React Hooks repo?

Reply:
Good point - added a "Next Steps" section